kpi/serializers/v2/service_usage.py

```diff
@@ -62,64 +80,180 @@ def get_storage_bytes(self, asset):
-        self._total_nlp_mt_characters = 0
+        self._total_nlp_asr_seconds_all_time = 0
+        self._total_nlp_asr_seconds_current_month = 0
+        self._total_nlp_asr_seconds_current_year = 0
+        self._total_nlp_mt_characters_all_time = 0
+        self._total_nlp_mt_characters_current_month = 0
+        self._total_nlp_mt_characters_current_year = 0
         self._total_storage_bytes = 0
         self._total_submission_count_all_time = 0
         self._total_submission_count_current_month = 0
+        self._total_submission_count_current_year = 0
+        self._current_month_start = None
+        self._current_year_start = None
         self._get_per_asset_usage(instance)
 
-    def get_total_nlp_asr_seconds(self, user):
-        return self._total_nlp_asr_seconds
+    def get_total_nlp_asr_seconds_all_time(self, user):
+        return self._total_nlp_asr_seconds_all_time
+
+    def get_total_nlp_asr_seconds_current_month(self, user):
+        return self._total_nlp_asr_seconds_current_month
+
+    def get_total_nlp_asr_seconds_current_year(self, user):
+        return self._total_nlp_asr_seconds_current_year
+
+    def get_total_nlp_mt_characters_all_time(self, user):
+        return self._total_nlp_mt_characters_all_time
+
+    def get_total_nlp_mt_characters_current_month(self, user):
+        return self._total_nlp_mt_characters_current_month
 
-    def get_total_nlp_mt_characters(self, user):
-        return self._total_nlp_mt_characters
+    def get_total_nlp_mt_characters_current_year(self, user):
+        return self._total_nlp_mt_characters_current_year
 
     def get_total_submission_count_all_time(self, user):
         return self._total_submission_count_all_time
 
     def get_total_submission_count_current_month(self, user):
         return self._total_submission_count_current_month
 
+    def get_total_submission_count_current_year(self, user):
+        return self._total_submission_count_current_year
+
     def get_total_storage_bytes(self, user):
         return self._total_storage_bytes
 
+    def get_current_month_start(self, user):
+        return self._current_month_start
+
+    def get_current_year_start(self, user):
+        return self._current_year_start
+
     def _get_per_asset_usage(self, user):
+        users = [user]
+        anchor_date, period_start, subscription_interval = None, None, None
+        # Get the organization ID passed in from the query parameters
+        organization_id = self.context['request'].query_params.get('organization_id')
+        if organization_id:
+            try:
+                organization = Organization.objects.filter(
+                    owner__organization_user__user=user,
+                    id=organization_id,
+                )
+                # If the user is in an organization, get all org users so we can query their total org usage
+                organization_users = OrganizationUser.objects.filter(organization=organization).select_related('user')
```

Comment:
I think you could reduce the code here by something like (pseudo code)

User.objects.filter(orgnaization__id=organization_id)

Often it's best to base the FROM part of the query on the table you actually want to return a list of, when possible at least. Then join that table to where you need to apply a filter.

The 404 may not be right here. I kind of don't care and this is pedantic, but a filter should simply filter the existing data. If that filter returns nothing. But if it's easier to raise either a 404 or 400 - I'm fine with it.